packages/visitor-keys/src/visitor-keys.ts

```diff
@@ -244,12 +244,7 @@ const additionalKeys: AdditionalKeys = {
   TSQualifiedName: ['left', 'right'],
   TSReadonlyKeyword: [],
   TSRestType: ['typeAnnotation'],
-  TSSatisfiesExpression: [
```

Comment:
Just noting for the record that it looks like this was set up this way from the start in https://github.com/typescript-eslint/typescript-eslint/pull/5717. The erroneous/misleading comment was added in https://github.com/typescript-eslint/typescript-eslint/pull/5926/files with no particular comment.

Just saying this all to do due diligence to investigate whether there was any particular reason in the paper trail that it was set up that way in the first place, but it doesn't seem there is (see @bradzacher's comment https://github.com/eslint/eslint/issues/19002#issuecomment-2402139894)

Reply:
I don't remember why I did this.
It was probably a cock up and I copied `TSTypeAssertion` rather than `TSAsExpression`.

If no tests fail then I assume it was defs just a bug 🤷